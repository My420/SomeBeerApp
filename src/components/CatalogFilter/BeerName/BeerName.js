import React from 'react';
import PropTypes from 'prop-types';
import styles from './BeerName.module.scss';
import {
  BEER_NAME_PROP,
  SEARCH_INPUT_MAX_LENGTH
} from '../../../utils/constants';

const BeerName = ({ value, onPropertyChange }) => {
  const onBeerNameChange = evt => {
    const name = evt.target.value;
    onPropertyChange(name, BEER_NAME_PROP);
  };

  return (
    <fieldset className={styles.nameField}>
      <legend className="visually-hidden">Beer Name</legend>
      <input
        className={styles.nameInput}
        type="text"
        value={value}
        placeholder="search"
        maxLength={SEARCH_INPUT_MAX_LENGTH}
        onChange={onBeerNameChange}
      />
    </fieldset>
  );
};

BeerName.propTypes = {
  value: PropTypes.string.isRequired,
  onPropertyChange: PropTypes.func.isRequired
};

export default BeerName;
