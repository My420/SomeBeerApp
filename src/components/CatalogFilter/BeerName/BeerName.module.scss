@import '../../../sass/variables.scss';

.nameField {
  border: none;
}

.nameInput {
  width: 300px;
  height: 30px;

  border: 2px solid $menu-bar-color;
  border-radius: 5px;
  @include input;
}

@media (max-width: 400px) {
  .nameInput {
    width: 220px;
  }
}
