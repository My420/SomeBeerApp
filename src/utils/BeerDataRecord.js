import { Record } from 'immutable';

const BeerDataRecord = Record(
  {
    id: null,
    name: null,
    tagline: null,
    first_brewed: null,
    description: null,
    image_url: null,
    abv: null,
    ibu: null,
    target_fg: null,
    target_og: null,
    ebc: null,
    srm: null,
    ph: null,
    attenuation_level: null,
    volume: {},
    boil_volume: {},
    method: {},
    ingredients: {},
    food_pairing: [],
    brewers_tips: null,
    contributed_by: null
  },
  'BeerDataRecord'
);

export default BeerDataRecord;
