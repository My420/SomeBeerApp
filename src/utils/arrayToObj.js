/* eslint no-param-reassign: "error" */

const arrayToObj = arr => {
  return arr.reduce((prevValue, element) => {
    prevValue[element.id] = element;
    return prevValue;
  }, {});
};

export default arrayToObj;
