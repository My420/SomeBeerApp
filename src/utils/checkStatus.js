const checkStatus = response => {
  const responseStatus = +response.status;
  if (responseStatus >= 200 && responseStatus < 300) {
    return response;
  }
  throw new Error(response.statusText);
};

export default checkStatus;
