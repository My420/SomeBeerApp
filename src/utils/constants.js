export const SHOP_NAME = 'Some Beer Shop';

export const LINK = {
  HOME: '/',
  CATALOG: '/catalog',
  ROULETTE: '/roulette',
  ABOUT: '/about',
  FAVORITE: '/favorite',
  CART: '/cart'
};

export const NAVIGATION_LINKS = [
  [LINK.HOME, 'Home'],
  [LINK.CATALOG, 'Catalog'],
  [LINK.ROULETTE, 'Roulette!'],
  [LINK.FAVORITE, 'Favorite'],
  [LINK.ABOUT, 'About'],
  [LINK.CART, 'Cart']
];

export const BROWSER_WIDTH_RATIO = {
  LPC: 4,
  MPC: 3,
  TABLET_9: 2,
  TABLET_7: 1,
  MOBILE: 0
};

export const BROWSER_MIN_WIDTH = {
  LPC: 1200,
  MPC: 993,
  TABLET_9: 768,
  TABLET_7: 480,
  MOBILE: 320
};

export const SOCIAL_LINK = {
  GITHUB: 'https://github.com/My420',
  EMAIL: 'mailto:klinovitsky.aleksey@gmail.com',
  SKYPE: 'skype:live:klinovitsky.aleksey?chat'
};

export const AUTHOR = 'Klinovitsky Aleksey';

export const POPULAR_BEER__GET_REQUEST = 'POPULAR_BEER__GET_REQUEST';
export const POPULAR_BEER__GET_SUCCESS = 'POPULAR_BEER__GET_SUCCESS';
export const POPULAR_BEER__GET_FAILURE = 'POPULAR_BEER__GET_FAILURE';

export const API_HOST = 'https://api.punkapi.com/v2/';

export const POPULAR_BEER_AMOUNT = 8;

export const LOAD_ERROR_MESSAGE =
  'Could not download data, please refresh the page';

export const ERROR_TITLE = 'Oops! Something went wrong.';
export const ERROR_ACTION = 'Please refresh page or try again later.';
