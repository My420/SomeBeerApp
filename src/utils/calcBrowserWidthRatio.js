import { BROWSER_MIN_WIDTH, BROWSER_WIDTH_RATIO } from './constants';

const calcBrowserWidthRatio = width => {
  if (width > BROWSER_MIN_WIDTH.LPC) {
    return BROWSER_WIDTH_RATIO.LPC;
  }
  if (width > BROWSER_MIN_WIDTH.MPC) {
    return BROWSER_WIDTH_RATIO.MPC;
  }
  if (width > BROWSER_MIN_WIDTH.TABLET) {
    return BROWSER_WIDTH_RATIO.TABLET_9;
  }
  if (width > BROWSER_MIN_WIDTH.MOBILE) {
    return BROWSER_WIDTH_RATIO.TABLET_7;
  }
  return BROWSER_WIDTH_RATIO.MOBILE;
};

export default calcBrowserWidthRatio;
