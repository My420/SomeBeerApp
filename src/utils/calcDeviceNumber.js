import { DEVICE_MIN_WIDTH, DEVICE_NUMBER } from './constants';

const calcDeviceNumber = width => {
  if (width > DEVICE_MIN_WIDTH.LPC) {
    return DEVICE_NUMBER.LPC;
  }
  if (width > DEVICE_MIN_WIDTH.MPC) {
    return DEVICE_NUMBER.MPC;
  }
  if (width > DEVICE_MIN_WIDTH.TABLET) {
    return DEVICE_NUMBER.TABLET;
  }
  return DEVICE_NUMBER.MOBILE;
};

export default calcDeviceNumber;
