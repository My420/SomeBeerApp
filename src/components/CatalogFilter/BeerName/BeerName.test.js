import React from 'react';
import { shallow } from 'enzyme';
import BeerName from './BeerName';
import { BEER_NAME_PROP } from '../../../utils/constants';

describe('test <BeerName /> component', () => {
  const mockFunc = jest.fn();
  const wrapper = shallow(
    <BeerName value="test" onPropertyChange={mockFunc} />
  );

  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should render input with correct value', () => {
    expect(wrapper.find('input').prop('value')).toBe('test');
  });

  test('should call onPropertyChange prop when input change ', () => {
    wrapper.find('input').simulate('change', { target: { value: 'Test' } });
    expect(mockFunc.mock.calls.length).toBe(1);
    expect(mockFunc.mock.calls[0][0]).toBe('Test');
    expect(mockFunc.mock.calls[0][1]).toBe(BEER_NAME_PROP);
  });
});
