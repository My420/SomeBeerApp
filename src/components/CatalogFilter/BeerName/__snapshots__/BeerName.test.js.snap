// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <BeerName /> component should match snapshot 1`] = `
<fieldset
  className="nameField"
>
  <legend
    className="visually-hidden"
  >
    Beer Name
  </legend>
  <input
    className="nameInput"
    maxLength={40}
    onChange={[Function]}
    placeholder="search"
    type="text"
    value="test"
  />
</fieldset>
`;
